pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -194,10 +194,15 @@ func (m *manager) AddContainer(pod *v1.Pod, containerID string) error {
 }
 
 func (m *manager) RemoveContainer(containerID string) error {
+	klog.Infof("[topologymanager] RemoveContainer - Container ID: %v", containerID)
+
 	podUIDString := m.podMap[containerID]
-	delete(m.podTopologyHints, podUIDString)
 	delete(m.podMap, containerID)
-	klog.Infof("[topologymanager] RemoveContainer - Container ID: %v podTopologyHints: %v", containerID, m.podTopologyHints)
+	delete(m.podTopologyHints[podUIDString], containerID)
```

Comment:
The old implementation allowed `RemoveContainer` to be called even when the `m.podTopologyHints[podUIDString]` didn't exist. Should we check here for existence before delete?